pygments/lexers/ssh.py

```diff
@@ -0,0 +1,200 @@
+            (r'\S+', String),
+        ],
+    }
+
+    def analyse_text(text):
+        if re.search(r'^[ \t]*(\w\S+[ \t]+)?(sk-)?(ecdsa|ssh)-\S+[ \t]+[A-Za-z0-9+/]+={0,2}', text):
+            return True
+
+
+class PublicKeyLexer(RegexLexer):
+    """
+    Lexer for SSH public key files.
+
+    .. versionadded:: 2.10
+    """
+    name = 'ssh_pubkey'
+    aliases = ['ssh_pubkey', 'sshpub']
+    filenames = ['id_*.pub']
+
+    flags = re.MULTILINE
+
+    tokens = {
+        'root': [
+            (r'^([ \t]*)'
+             r'(?P<keytype>\S+)([ \t]+)'
+             r'(?P<keydata>\S+)'
+             r'(?:([ \t]+)(?P<keycomment>\S.*?))?'
+             r'(\s*)$',
+             bygroups(
+                 Whitespace,
+                 Keyword.Type,  # keytype
+                 Whitespace,
+                 String,  # keydata
+                 Whitespace,
+                 Comment.Special,  # keycomment
+                 Whitespace
+             )),
+        ]
+    }
+
+    def analyse_text(text):
+        match = re.search(r'^[ \t]*(sk-)?(ecdsa|ssh)-\S+[ \t]+[A-Za-z0-9+/]+={0,2}(?P<keycomment>[ \t]+\S)?', text)
+        if match:
+            # 0.9: pubkeys are more likely to contain key comments than known hosts files, 0.8: tie with known hosts
+            return 0.9 if match.group('keycomment') else 0.8
+
+
+class KnownHostsLexer(RegexLexer):
+    """
+    Lexer for SSH known hosts files.
+
+    .. versionadded:: 2.10
+    """
+    name = 'known_hosts'
+    aliases = ['known_hosts']
+    filenames = ['known_hosts', 'known_hosts2']
+
+    flags = re.MULTILINE
+
+    tokens = {
+        'root': [
+            (r'^(\s*)(#.*?)(\s*)$', bygroups(Whitespace, Comment, Whitespace)),  # comments only at beginning of line
+            (r'^([ \t]*)'
+             r'(?:(?P<marker>@\S+)([ \t]+))?'
+             r'(?P<hostnames>\S+)([ \t]+)'
+             r'(?P<keytype>(?:sk-)?(?:ecdsa|ssh)-\S+)([ \t]+)'
+             r'(?P<keydata>\S+)'
+             r'(?:([ \t]+)(?P<keycomment>\S.*?))?'
+             r'(\s*)$',
+             bygroups(
+                 Whitespace,
+                 Name.Class,  # marker
+                 Whitespace,
+                 Name.Entity,  # hostnames
+                 Whitespace,
+                 Keyword.Type,  # keytype
+                 Whitespace,
+                 String,  # keydata
+                 Whitespace,
+                 Comment.Special,  # keycomment
+                 Whitespace,
+             )),
+        ]
+    }
+
+    def analyse_text(text):
+        match = re.search(r'^[ \t]*(?P<marker>@[\w-]+[ \t]+)?\S+[ \t]+(sk-)?(ecdsa|ssh)-\S+[ \t]+[A-Za-z0-9+/]+={0,2}', text)
+        if match:
+            # 0.8: tie with pubkeys
+            return True if match.group('marker') else 0.8
+
+
+class SshConfigLexer(RegexLexer):
+    """
+    Lexer for SSH client config files.
+
+    .. versionadded:: 2.10
+    """
+    name = 'SSH client config'
+    aliases = ['ssh_config']
+    filenames = ['ssh_config', 'config']
```

Comment:
`config` is pretty generic, don't know if we see that as a problem here.

Reply:
Nobody else is claiming config yet but I don't think this lexer should claim it. It feels too generic -- please remove it.

Reply:
Done.